src/osd/osd_types.h

```diff
@@ -2962,6 +2962,11 @@ struct pg_log_t {
   mempool::osd::list<pg_log_entry_t> log;  // the actual log.
   
   pg_log_t() = default;
+  pg_log_t(const pg_log_t& other) = default;
+  pg_log_t& operator=(const pg_log_t& other) = default;
+  pg_log_t(pg_log_t&& other) = default;
+  pg_log_t& operator=(pg_log_t&& other) = default;
+
```

Comment:
sorry if i'm rusty on this, but i'm looking at https://en.wikipedia.org/wiki/Special_member_functions. the non-default constructor does prevent the compiler from generating a default constructor, but it doesn't appear to have any effect on the move operations. are you certain that those move operations aren't being generated already?